### PR TITLE
Update Mikado to 2.0 RC1

### PR DESCRIPTION
**Pull request by lucventurini**
Updating mikado (https://github.com/EI-CoreBioinformatics/mikado) to version 2.0 RC1.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by lucventurini** [comment]
Dear @bioconda/core, 

Apologies for asking for help. Unfortunately I seem to be unable to make this recipe work, however much I try: the build constantly fails at the cleanup stage, complaining that "README.md" cannot be deleted due to a "Permission denied" error. I cannot for the life of me understand why that would be the case - for what is worth, the `README.md` file in my project is not write-protected.

If you could please help, I would be grateful.

Kind regards

**Comment by lucventurini** [comment]
@bioconda/bot please add label help wanted

**Comment by lucventurini** [comment]
Dear @dpryan79 , @bioconda/core 

The problem was caused by the unit-tests changing the permissions of the files in the folder. This is unexpected and apparently happened only on one of the runs. I suspect the origin might be in `numpy` as I use its `pytester` functions.

If you could review and merge into `bioconda` I would be very grateful for your time!

File: recipes/mikado/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+python:
+  - 3.6
+  - 3.7
+  - 3.8
+numpy:
+  - 1.17
+  - 1.18
```
Comment (dpryan79) [inline recipes/mikado/conda_build_config.yaml:7]: Can we get rid of this? We'll start building for python 3.8 soon and numpy can be restricted in the meta.yaml.

File: recipes/mikado/meta.yaml [hunk 2]
```diff
@@ -1,82 +1,75 @@
 {% set name = "mikado" %}
-{% set version = "1.2.4" %}
+{% set version = "2.0rc1" %}
```
Comment (dpryan79) [inline recipes/mikado/meta.yaml:2]: We don't normally ship release candidates.

File: recipes/mikado/meta.yaml [hunk 3]
```diff
@@ -1,82 +1,75 @@
...
+  # {{ PYTHON }} setup.py install --single-version-externally-managed --record record.txt
+  script: chmod -R a+rw *; {{ PYTHON }}  -m pip install --no-deps --ignore-installed -v .; chmod -R a+rw *
 
 requirements:
   build:
     - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
   host:
-    - python
+    - python {{ python }}
```
Comment (dpryan79) [inline recipes/mikado/meta.yaml:30]: Please get rid of `{{ python }}`, it's redundant. That goes for `{{ numpy }}` too

File: recipes/mikado/meta.yaml [hunk 4]
```diff
@@ -96,15 +89,14 @@ test:
...
     - Mikado.utilities
-    - nose
-  commands:
-    - mikado --help
-    - daijin --help
-    - python -c "import Mikado; Mikado.test()"
+  # commands:
+    # - mikado --help
+    # - daijin --help
+    # - python -c "import Mikado; Mikado.test()"
```
Comment (dpryan79) [inline recipes/mikado/meta.yaml:95]: Please remove if not needed

File: recipes/mikado/conda_build_config.yaml (hunk 1 above)
Comment (lucventurini) [reply recipes/mikado/conda_build_config.yaml:7]: Yes, I introduced the file when trying to understand the reason of the test failures. I can remove it now.

File: recipes/mikado/meta.yaml (hunk 3 above)
Comment (lucventurini) [reply recipes/mikado/meta.yaml:30]: Will do

**Comment by lucventurini** [comment]
Dear @dpryan79 

I have amended the recipe according to your suggestions.
Regarding the shipping of a release candidate in BioConda, our reasons are twofold:

- we are planning to make a paper out of the new version, as we made many changes since the original version of the code (almost three years ago). We would like the versions we use for the analysis to be available here, for full reproducibility.
- Relatedly, many users have asked on our github page which version they should install, given 
  a- the presence of many bugs/performance that have been fixed in the `master` trunk since the last stable release (~2 years ago)
  b- the much easier installation process through Conda.

We do undestand of course that this might break BioConda release policy. If that is the case, we might decide for an alternative shipping strategy to be in place.

Kind regards

**Comment by dpryan79** [comment]
I'm not sure why you're labeling it as a release candidate if you're writing a paper using the results, but I guess that's up to you. I'll push the package out.

**Comment by dpryan79** [comment]
@bioconda-bot Please fetch artifacts

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Comment by lucventurini** [comment]
> I'm not sure why you're labeling it as a release candidate if you're writing a paper using the results, but I guess that's up to you. I'll push the package out.

We foresee that reviewers might ask for minor revisions and/or we might make some minor adjustments before the project is finished. We would like Mikado 2.0 to be the release with which the article is published (rather than 2.0.1 or some such).

Many thanks for merging this!

**Pull request merged**